src/main/java/org/deepsymmetry/beatlink/data/OpusProvider.java

```diff
@@ -65,11 +71,33 @@ public boolean isRunning() {
      */
     private final AtomicReference<FileSystem> usb2filesystem = new AtomicReference<>();
 
+    /**
+     * Storing the current USB slot that the Opus is reporting as USB2. The Opus can report the same slot differently
+     * when it restarts so it needs to be re-evaluated when PSSI + track ID don't match.
+     */
+    private AtomicReference<CdjStatus.TrackSourceSlot> opusReportedDataSlot2 =  new AtomicReference<>();
```

Comment:
Do we even need this information? I thought we were just going to check track ID and PSSI, and ignore these bytes which seem unreliable and poorly understood?

Reply:
We do in this iteration. I am going to do the work to add the third USB in another PR to keep things small, there will be a bit of a refactoring to get there.

Reply:
Can you explain how? I completely don’t understand how we care what slot we think the status packet is talking about if we are cross-checking against the PSSI and track ID.

Reply:
This value holds the TrackSourceSlot that the opus is reporting as the source slot (could be USB or SD, it is determined on startup of the opus). When we receive the PSSI, we compare it against the two drives and set this value based on which one matches. 

We then use this saved TrackSourceSlot to compare against the slot that the opus reports in the CdjStatus packet, which will be the same as the one reported in the PSSI packet. This happens in `findDatabase` and `findFilesystem` methods. 

```quote
    public FileSystem findFilesystem(SlotReference slot) {
        CdjStatus.TrackSourceSlot slot1 = opusReportedDataSlot1.get();

        if (slot.slot == slot1) {
            return usb1filesystem.get();
        }

        CdjStatus.TrackSourceSlot slot2 = opusReportedDataSlot2.get();

        if (slot.slot == slot2) {
            return usb2filesystem.get();
        }

        return null;
    }
```

Reply:
I see. I don’t like this, I don’t trust the value will be stable, so we should be ignoring it. But I agree that we can remove this as part of the broader refactoring necessary to support all three slots. In that world we must ignore these values anyway. So I will resolve this for now, pending your next PR.

Reply:
In theory, none of that is needed though. I should be able to handle this without knowing what the Opus is reporting as the slot.

Reply:
Exactly, we’re kind of saying the same thing here, and that it can wait for your next PR. 😄